testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/websocket/WebSocketTestCase.java

```diff
@@ -40,17 +39,16 @@ public static WebArchive deploy() {
                                 new PropertyPermission("node0", "read"),
                                 new PropertyPermission("jboss.http.port", "read"),
                                 // Needed for the serverContainer.connectToServer()
-                                new SocketPermission("*:" + TestSuiteEnvironment.getHttpPort(), "connect,resolve")
+                                new SocketPermission(TestSuiteEnvironment.getServerAddress() + ":" + TestSuiteEnvironment.getHttpPort(), "connect,resolve"),
```

Comment:
This is not a correct fix. The deployment doesn't require any permission. We should not run the test within the deployment, the `@RunAsClient` should be rather used. 

Use sth like this:

```java
@RunWith(Arquillian.class)
@RunAsClient
public class WebSocketTestCase {

    @Deployment(testable=false)
    public static WebArchive deploy() {
        return ShrinkWrap.create(WebArchive.class, "websocket.war")
                .addClasses(AnnotatedEndpoint.class)
                .addAsManifestResource(new StringAsset("io.undertow.websockets.jsr.UndertowContainerProvider"),
                        "services/javax.websocket.ContainerProvider");
    }

    @Test
    public void testWebSocket() throws Exception {
        AnnotatedClient endpoint = new AnnotatedClient();
        WebSocketContainer serverContainer = ContainerProvider.getWebSocketContainer();
        try (Session session = serverContainer.connectToServer(
                endpoint,
                new URI("ws", "", TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getHttpPort(),
                        "/websocket/websocket/Stuart", "", ""))) {
            Assert.assertEquals("Hello Stuart", endpoint.getMessage());
        }
    }
}
```